controllers/provisioners/eks/helpers.go

```diff
@@ -1173,3 +1173,40 @@ func (ctx *EksInstanceGroupContext) GetDesiredMixedInstancesPolicy(name string)
 
 	return policy
 }
+
+func FilterSupportedArch(architectures []string) string {
```

Comment:
Is `FilterSupportedArch` only meant to check if a slice of architectures is in slice of `SupportedArchitectures` ?
Could we just use `common.StringSliceContains`, or a different generic function?
In the below code it seems we return the first architecture that we find a match for, is there a condition where there are multiple supported architectures?

Reply:
`FilterSupportedArch` returns the first match between the `SupportedArchitectures`
from my investigation of the aws api most instance types either return 
```
"arch": [
    "i386",
    "x86_64"
  ]
```
or
```
"arch": [
    "arm64"
  ]
```

since i386 isn't a supported arch, `common.StringSliceContains` would fail. Happy to use a different generic if one exists. If ever amazon provides an instance with multiple supported architectures, we would either need to explain the preference order, or provide some way of preferencing one.